src/Model/test.js

```diff
@@ -208,6 +210,88 @@ describe('CRUD', () => {
     expect(nextObserver).toHaveBeenCalledTimes(1)
     expect(completionObserver).toHaveBeenCalledTimes(1)
   })
+  it('can destroy a record and its children permanently', async () => {
+    const { database, projects, tasks, comments } = mockDatabase()
+
+    const project = await projects.create(mock => {
+      mock.name = 'foo'
+    })
+
+    const task = await tasks.create(mock => {
+      mock.project.set(project)
+    })
+
+    const comment = await comments.create(mock => {
+      mock.task.set(task)
+    })
+
+    database.adapter.batch = jest.fn()
+    const spyBatchDB = jest.spyOn(database, 'batch')
+
+    const spyOnPrepareDestroyPermanentlyProject = jest.spyOn(project, 'prepareDestroyPermanently')
+    const spyOnPrepareDestroyPermanentlyTask = jest.spyOn(task, 'prepareDestroyPermanently')
+    const spyOnPrepareDestroyPermanentlyComment = jest.spyOn(comment, 'prepareDestroyPermanently')
+
+    await project.experimentalDestroyPermanently()
+
+    expect(spyOnPrepareDestroyPermanentlyProject).toHaveBeenCalledTimes(1)
+    expect(spyOnPrepareDestroyPermanentlyTask).toHaveBeenCalledTimes(1)
+    expect(spyOnPrepareDestroyPermanentlyComment).toHaveBeenCalledTimes(1)
+
+    expect(spyBatchDB).toHaveBeenCalledWith(comment, task, project)
+  })
+  it('can mark a record as deleted', async () => {
+    const database = makeDatabase()
+    database.adapter.batch = jest.fn()
+    const spyBatchDB = jest.spyOn(database, 'batch')
+
+    const collection = database.collections.get('mock')
+
+    const m1 = await collection.create()
+
+    const spyOnMarkAsDeleted = jest.spyOn(m1, 'prepareMarkAsDeleted')
+    const nextObserver = jest.fn()
+    const completionObserver = jest.fn()
+    m1.observe().subscribe(nextObserver, null, completionObserver)
+
+    await m1.markAsDeleted()
+
+    expect(spyBatchDB).toHaveBeenCalledWith(m1)
```

Comment:
this isn't right, you should check it after create, not after markAsDeleted - it looks as if you were checking that calling markAsDeleted sends the record to batch

Reply:
fixed